example/cripts/example1.cc

```diff
@@ -0,0 +1,242 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// The primary include file, this has to always be included
+#include <cripts/Preamble.hpp>
+
+#include <cripts/Bundles/Common.hpp>
+
+// Globals for this Cript
+static Matcher::Range::IP CRIPT_ALLOW({"192.168.201.0/24", "10.0.0.0/8"});
+
+// This is called only when the plugin is initialized
+do_init()
+{
+  TSDebug("Cript", "Hello, example1 plugin is being initialized");
+}
+
+do_create_instance()
+{
+  instance.metrics[0] = Metrics::Sum::create("cript.example1.c0");
+  instance.metrics[1] = Metrics::Sum::create("cript.example1.c1");
+  instance.metrics[2] = Metrics::Sum::create("cript.example1.c2");
+  instance.metrics[3] = Metrics::Sum::create("cript.example1.c3");
+  instance.metrics[4] = Metrics::Sum::create("cript.example1.c4");
+  instance.metrics[5] = Metrics::Sum::create("cript.example1.c5");
+  instance.metrics[6] = Metrics::Sum::create("cript.example1.c6");
+  instance.metrics[7] = Metrics::Sum::create("cript.example1.c7");
+  instance.metrics[8] = Metrics::Sum::create("cript.example1.c8"); // This one should resize() the storage
+
+  Bundle::Common::activate().dscp(10).cache_control("max-age=259200");
+}
+
+do_txn_close()
+{
+  borrow conn = Client::Connection::get();
+
+  conn.pacing = Cript::Pacing::Off;
+  CDebug("Cool, TXN close also works");
+}
+
+do_send_request()
+{
+  borrow req = Server::Request::get();
+
+  req["X-Leif"] = "Meh";
+}
+
+do_read_response()
+{
+  borrow resp = Server::Response::get();
+
+  resp["X-DBJ"] = "Vrooom!";
+}
+
+do_send_response()
+{
+  borrow resp = Client::Response::get();
+  borrow conn = Client::Connection::get();
+  string msg  = "Eliminate TSCPP";
+
+  resp["Server"]         = "";        // Deletes the Server header
```

Comment:
I don't think we can assume that an empty value implies field deletion. It's valid for an HTTP field to have an empty value. `Accept-Encoding`, for instance, can have an empty value and it has a particular meaning:

https://www.rfc-editor.org/rfc/rfc2616#section-14.3

> If the Accept-Encoding field-value is empty, then only the "identity" encoding is acceptable.

I suspect we need to have a `.delete()` function or something like it.

Reply:
You already have it :). 

```
  resp["Good-Brian"].clear();
```

or

```
  resp.erase("Good-Brian");
```

Reply:
Cool. Let's update the comment then - it led me to believe that setting to an empty string was overloaded to delete the field.